pkg/network/protocols/http/shared_libraries.go

```diff
@@ -294,6 +295,35 @@ func (r *soRegistry) Unregister(pid uint32) {
 	delete(r.byPID, pid)
 }
 
+func quircksWaitLib(hostlibPath string) error {
+	// java netty when injecting libnetty_tcnative_linux_.*.so would copy the file (from his jar) to a workdir
+	// then inject it to the process via System.Loadlibrary()
+	// Usually we got the event when the file is just created and contain 0 bytes
+	if strings.Contains(hostlibPath, "libnetty_tcnative_linux_") {
+		s, err := os.Stat(hostlibPath)
+		if err != nil {
+			return err
+		}
+		time.Sleep(time.Millisecond)
+		libSize := s.Size()
+		end := time.Now().Add(100 * time.Millisecond)
+		for !time.Now().After(end) {
+			s, err = os.Stat(hostlibPath)
+			if err != nil {
+				return err
+			}
+			if s.Size() != libSize || s.Size() == 0 {
+				libSize = s.Size()
+				time.Sleep(5 * time.Millisecond)
+				continue
+			}
+			return nil
+		}
+		return fmt.Errorf("timeout current size %d", libSize)
```

Comment:
the code here is problematic 
1. having libSize != 0 does not guarantee you have the full lib written
2. Using sleep here is very patchy. We must find a better solution

Reply:
furthermore, the sleep here, blocks all events to soResigtry

Reply:
it was more a hack when I tried to recover/copy the lib before to be deleted
I move the registration logic asynchronously